src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -248,6 +251,19 @@ public virtual void Execute()
                     Test.RequiresThread ? "RequiresThread" : "different Apartment");
 
 #if APARTMENT_STATE
+#if NETSTANDARD2_0
+                if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+                {
+                    if(targetApartment != ApartmentState.Unknown)
```

Comment:
Could you do this check the other way around (targetApartment field first, then `IsOSPlatform`)?

Also, we need a space after the `if` keyword.

CI is still hanging except on Windows. Can you try using the check `targetApartment == ApartmentState.STA` instead of `targetApartment != ApartmentState.Unknown`?